rechunker/api.py

```diff
@@ -10,12 +10,100 @@
 from rechunker.algorithm import rechunking_plan
 
 
-def rechunk_zarr2zarr_w_dask(
+def _shape_dict_to_tuple(dims, shape_dict):
+    # convert a dict of shape
+    shape = [shape_dict[dim] for dim in dims]
+    return tuple(shape)
+
+
+def _get_dims_from_zarr_array(z_array):
+    # use Xarray convention
+    # http://xarray.pydata.org/en/stable/internals.html#zarr-encoding-specification
+    return z_array.attrs["_ARRAY_DIMENSIONS"]
+
+
+def _zarr_empty(shape, store_or_group, chunks, dtype, name=None):
+    # wrapper that maybe creates the array within a group
+    if name is not None:
+        assert isinstance(store_or_group, zarr.hierarchy.Group)
+        return store_or_group.empty(name, shape=shape, chunks=chunks, dtype=dtype)
+    else:
+        return zarr.empty(shape, chunks=chunks, dtype=dtype, store=store_or_group)
+
+
+def rechunk(
+    source, target_chunks, max_mem, target_store, temp_store=None,
+):
+    """
+    Rechunk a Zarr Array or Group
+
+    Parameters
+    ----------
+    source : zarr.Array or zarr.Group
+    target_chunks : tuple or dict
+        The desired chunks of the array after rechunking.
```

Comment:
Might be nice to specify that tuples are for Array, dicts are for Groups (IIUC).

Reply:
Yes, I'll do this as part of a broader doc update.

Part of the complexity is that we also support dicts for arrays, if the arrays have dimensions, e.g. `target_chunks={'x': 10, 'y': 5}`. The group version of this would be
```python
{'foo': {'x': 10, 'y': 5},
 'bar': {'x': 5, 'y': 1}
```